### DISCUSSION TITLE
FR: Automatic MA device reload on new device addition

### DISCUSSION THREAD

**Discussion by Julien-Decoen** [jseidl/magic-areas#417] [Ideas / Feature Requests]
Hi @jseidl ,

I’d like to propose a new feature for Magic Areas: automatic device reload when a new device is added to an area.

Currently, when a new device is added to an area, we need to manually reload the Magic Areas device or restart Home Assistant to ensure the new device is taken into account. With this new feature, Magic Areas would automatically refresh itself whenever a new device is added to an area, making the process more seamless and eliminating the need for manual reloads.

This could greatly improve the user experience by ensuring that newly added devices are instantly recognized and integrated into the area's logic without any extra steps.

What are your thoughts on this potential feature?

Julien

**Comment by jseidl** [comment]
Sounds like a great idea. Will add

**Comment by jseidl** [comment]
Good news! https://github.com/jseidl/hass-magic_areas/pull/422 is coming out

**Comment by jseidl** [comment]
Implemented on https://github.com/jseidl/hass-magic_areas/pull/422
